instrumentation/net_http/test/opentelemetry/instrumentation/net/http/dup/instrumentation_test.rb

```diff
@@ -0,0 +1,366 @@
+      )
+      assert_requested(
+        :get,
+        'https://example.com/timeout',
+        headers: { 'Traceparent' => "00-#{span.hex_trace_id}-#{span.hex_span_id}-01" }
+      )
+    end
+
+    it 'merges http client attributes' do
+      OpenTelemetry::Common::HTTP::ClientContext.with_attributes('peer.service' => 'foo', 'http.target' => 'REDACTED', 'url.path' => 'REDACTED') do
+        Net::HTTP.get('example.com', '/success')
+      end
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'GET'
+      _(span.attributes['http.method']).must_equal 'GET'
+      _(span.attributes['http.scheme']).must_equal 'http'
+      _(span.attributes['http.status_code']).must_equal 200
+      _(span.attributes['http.target']).must_equal 'REDACTED'
+      _(span.attributes['net.peer.name']).must_equal 'example.com'
+      _(span.attributes['net.peer.port']).must_equal 80
+      _(span.attributes['peer.service']).must_equal 'foo'
+      # stable semantic conventions
+      _(span.attributes['http.request.method']).must_equal 'GET'
+      _(span.attributes['url.scheme']).must_equal 'http'
+      _(span.attributes['http.response.status_code']).must_equal 200
+      _(span.attributes['url.path']).must_equal 'REDACTED'
+      _(span.attributes['server.address']).must_equal 'example.com'
+      _(span.attributes['server.port']).must_equal 80
+      _(span.attributes['peer.service']).must_equal 'foo'
+      assert_requested(
+        :get,
+        'http://example.com/success',
+        headers: { 'Traceparent' => "00-#{span.hex_trace_id}-#{span.hex_span_id}-01" }
+      )
+    end
+
+    it 'merges http client attribute url query' do
+      Net::HTTP.get('example.com', '/success?hello=there')
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'GET'
+      _(span.attributes['url.query']).must_equal 'hello=there'
+      _(span.attributes['url.path']).must_equal '/success'
+      assert_requested(
+        :get,
+        'http://example.com/success?hello=there',
+        headers: { 'Traceparent' => "00-#{span.hex_trace_id}-#{span.hex_span_id}-01" }
+      )
+    end
+  end
+
+  describe 'untraced?' do
+    before do
+      stub_request(:get, 'http://example.com/body').to_return(status: 200)
+      stub_request(:get, 'http://foobar.com/body').to_return(status: 200)
+      stub_request(:get, 'http://bazqux.com/body').to_return(status: 200)
+    end
+
+    describe 'untraced_hosts option' do
+      before do
+        instrumentation.instance_variable_set(:@installed, false)
+        config = {
+          untraced_hosts: ['foobar.com', /bazqux\.com/]
+        }
+
+        instrumentation.install(config)
+      end
+
+      it 'does not create a span when request ignored using a string' do
+        Net::HTTP.get('foobar.com', '/body')
+        _(exporter.finished_spans.size).must_equal 0
+      end
+
+      it 'does not create a span when request ignored using a regexp' do
```

Comment:
@hannahramadan - The Net::HTTP failing test looks related to changes from a recent PR. Those fixes may need to be incorporated into the dup/old/stable tests on this one: 
https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1604/files#diff-99a3ac2b3a9bb5c2d057a09c5101ddb9d5b472344b41269b8a9f1269bceabaaa